### ISSUE TITLE
improve space detection and remove pdfminer high level code

### ISSUE THREAD

**Issue by metebalci** [metebalci/pdftitle#25]
Text in the PDF file might not contain space character but the space might be indicated with an actual (additional) horizontal position difference between the glyphs before and after the space, so between the last char and the first char of the words. pdfminer has a high level code detecting this i.e. if the space between chars is greater than a certain threshold (possibly specified in the font file). It is better to do this manually and also implement spacing if vertical positions also changed (title in more than one lines). When this is done, I think, the get_title_from_io method can be simplified by removing the TextConverter and PDFPageInterpreter related parts.

**Comment by mdbraber** [comment]
Is it currently expected if a headline spans multiple lines it will fail to output the right format (in my case: words on different lines are joined without spaces)?

**Comment by mdbraber** [comment]
Figured out what's happening in the current version. I've got PDF titles split over multiple lines, but the lines itself hold spaces, so the statement on line 564 (`if " " not in title`) doesn't return True. When forcing this it works in my case. Maybe possibly add an argument to force space correction (or just alway correct spaces)?

**Comment by metebalci** [comment]
Yes it makes sense to add an argument. If possible, can you share the pdf
so it can be used to validate this improvement ?

**Comment by mdbraber** [comment]
I've sent the PDF files via e-mail for validation. It works now on some, but not yet on all articles when forcing space correction.
